gnomad_qc/v4/create_release/create_release_sites_ht_exomes.py

```diff
@@ -0,0 +1,668 @@
+                for attr in val.split("."):
+                    ht = ht[attr]
+                select_fields[key] = ht
+    return select_fields
+
+
+def get_ht(dataset: str, _intervals, test, release_exists) -> hl.Table:
+    """
+    Return the appropriate hail table with selects applied.
+
+    :param dataset: Hail Table to join.
+    :param _intervals: Intervals for reading in hail Table.
+    :param test: Whether call is for a test run.
+    :return: Hail Table with fields to select.
+    """
+    logger.info("Getting the %s dataset and its selected annotations...", dataset)
+    config = get_config(release_exists=release_exists)[dataset]
+
+    # There is no single path for insilico predictors, so we need to handle this case
+    # separately.
+    if dataset == "in_silico":
+        base_ht = config["ht"]
+    else:
+        ht_path = config["path"]
+        logger.info("Reading in %s", dataset)
+        base_ht = hl.read_table(ht_path, _intervals=_intervals)
+
+    if test:
+        base_ht = hl.filter_intervals(
+            base_ht,
+            [
+                hl.parse_locus_interval(
+                    "chr1:55039447-55064852", reference_genome="GRCh38"
+                )
+            ],
+        )
+
+    select_fields = get_select_fields(config.get("select"), base_ht)
+
+    if "custom_select" in config:
+        custom_select_fn = config["custom_select"]
+        select_fields = {**select_fields, **custom_select_fn(base_ht)}
+
+    if "field_name" in config:
+        field_name = config.get("field_name")
+        select_query = {field_name: hl.struct(**select_fields)}
+    else:
+        select_query = select_fields
+
+    return base_ht.select(**select_query)
+
+
+def join_hts(
+    base_table: hl.Table,
+    tables: List[str],
+    new_partition_percent: float,
+    test: bool,
+    release_exists: bool,
+) -> hl.Table:
+    """
+    Outer join a list of Hail Tables.
+
+    :param base_table: Dataset to use for interval partitioning.
+    :param tables: List of tables to join.
+    :param new_partition_percent: Percent of base_table partitions used for final
+        release Hail Table.
+    :param test: Whether this is for a test run.
+    :param release_exists: Whether the release HT already exists.
+    :return: Hail Table with datasets joined.
+    """
+    logger.info(
+        "Reading in %s to determine partition intervals for efficient join",
+        base_table,
+    )
+    base_ht_path = get_config()[base_table]["path"]
+    base_ht = hl.read_table(base_ht_path)
+    if test:
+        # Filter to PCSK9 for testing.
+        base_ht = hl.filter_intervals(
+            base_ht,
+            [
+                hl.parse_locus_interval(
+                    "chr1:55039447-55064852", reference_genome="GRCh38"
+                )
+            ],
+        )
+    partition_intervals = base_ht._calculate_new_partitions(
+        base_ht.n_partitions() * new_partition_percent
+    )
+
+    # Reorg list so base table is first.
```

Comment:
this assumes that the base table is part of the list of tables to join, which it should be by default, maybe it's worth adding a warning / check or note to the docstring about this?